src/main/java/com/lambdatest/tunnel/Tunnel.java

```diff
@@ -584,58 +549,57 @@ public void runCommand(String command) throws IOException {
-                                if (br.readLine() != null) {
-                                    tunnelFlag = true;
-                                    System.out.println("Tunnel Started Successfully");
-                                    break;
+    private void waitForTunnelToStart() throws IOException {
+        RequestConfig config = RequestConfig.custom()
+                .setConnectTimeout(HTTP_TIMEOUT)
+                .setSocketTimeout(HTTP_TIMEOUT)
+                .setConnectionRequestTimeout(HTTP_TIMEOUT)
+                .build();
+        try(CloseableHttpClient httpClient = HttpClientBuilder.create().setDefaultRequestConfig(config).build()){
+            for (int retryCount = 0; retryCount < MAX_TUNNEL_STARTUP_RETRY_COUNT; retryCount++) {
+                try {
+                    Thread.sleep(TUNNEL_INITIAL_WAIT_AND_RETRY_BACKOFF);
+                    System.out.println("Checking Tunnel Status");
+                    String infoAPIGetEndpoint = "http://127.0.0.1:" + Integer.toString(infoAPIPortValue) + TUNNEL_INFO_API;
+                    HttpGet httpGet = new HttpGet(infoAPIGetEndpoint);
+                    try(CloseableHttpResponse execute = httpClient.execute(httpGet)){
+                        if (execute.getStatusLine().getStatusCode() == 200) {
+                            System.out.println("Received Tunnel Status Response");
+                            BufferedReader reader = new BufferedReader(new InputStreamReader(execute.getEntity().getContent()));
+                            StringBuilder response = new StringBuilder();
+                            String line;
+                            while ((line = reader.readLine()) != null) {
+                                response.append(line);
+                            }
+                            JSONObject tunnelInfo = new JSONObject(response.toString());
+                            if (tunnelInfo.getString("status").equals("SUCCESS")) {
+                                tunnelFlag = true;
+                                System.out.println("Tunnel Started Successfully");
+                                if (tunnelInfo.getJSONObject("data") != null && tunnelInfo.getJSONObject("data").has("id")) {
+                                    tunnelID = tunnelInfo.getJSONObject("data").getInt("id");
                                 }
+                                System.out.println("Tunnel ID: " + tunnelID);
+                                break;
+                            } else {
+                                System.out.println("Tunnel Status: " + tunnelInfo.getString("status"));
                             }
-                        } catch (Exception e) {
-                            // System.out.println("Not found any file");
+                        } else {
+                            System.out.println("Tunnel Status Response: " + execute.getStatusLine().getStatusCode());

```

Comment:
I think we can log this one to the logger instead of console. Because in the first a few attemps tunnel may not be ready and we will see this message in the console directly.
`Tunnel Status Response: 500`
Customer can think there is a problem with tunnel starting.

Reply:
Agree..there is no logger however..

Reply:
I thought there is already a logger. Ok then we can ignore